targets/TARGET_STM/TARGET_STM32F2/gpio_irq_api.c

```diff
@@ -304,7 +304,7 @@ void gpio_irq_set(gpio_irq_t *obj, gpio_irq_event event, uint32_t enable)
                 mode = STM_MODE_IT_FALLING;
                 obj->event = EDGE_FALL;
             } else { // NONE or RISE
-                mode = STM_MODE_IT_EVT_RESET;
+                mode = STM_MODE_INPUT;
```

Comment:
No mistake :-)
This mode is sent to ST HAL, but STM_MODE_IT_EVT_RESET is not part of GPIO mode values...
I then set the default value.
(issue was seen with MBED_A7 test)